### PR TITLE
coqPackages.serapi: init at 8.14.0+0.14.0 for Coq 8.14 & OCaml < 4.12

### PR DESCRIPTION
**Pull request by vbgl**
###### Motivation for this change

https://github.com/NixOS/nixpkgs/pull/147249#issuecomment-983650084

Tested that `(mkCoqPackages (coq_8_14.override { customOCamlPackages = ocaml-ng.ocamlPackages_4_10; })).serapi` builds.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/coq-modules/serapi/default.nix [hunk 1]
```diff
@@ -27,7 +28,11 @@ in
   pname = "serapi";
   inherit version release;
 
-  defaultVersion =  with versions; switch coq.version [
+  defaultVersion =  with versions;
+    if isEq "8.14" coq.version && isLt "4.12" coq.ocamlPackages.ocaml.version
+    then "8.14.0+0.14.0"
+    else
```
Comment (CohenCyril) [inline pkgs/development/coq-modules/serapi/default.nix:34]: @vbgl why don't you integrate this to the switch below?

Comment (vbgl) [reply pkgs/development/coq-modules/serapi/default.nix:34]: Not sure how to do it. But your suggestion is welcome.

Comment (vbgl) [reply pkgs/development/coq-modules/serapi/default.nix]: I’ve just proposed an other version. Thanks.

Comment (CohenCyril) [reply pkgs/development/coq-modules/serapi/default.nix]: Looks better indeed

**Pull request merged**